### ISSUE TITLE
[UX] Search: make it clear what the 're-index site' button does

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1321]
Re-index site sounds like something fun and friendly, and people often click this button not realzing they are about to do damage.  Let's warn people appropriately.

Tasks:
- Change text to more clearly indicate what will happen when the button is clicked
- Change button color to red

**Comment by klonos** [comment]
Still, we don't want to unnecessarily scare people. So, yes, change the button color to red, but be a bit more verbose along these lines "Delete current search index (the index will be recreated automatically over time/cron runs)."

**Comment by jenlampton** [comment]
Okay, how about 'Invalidate search index' for the button text.

A more detailed explanation is provided in the text on the confirm form:
`The search index is not cleared but will be systematically updated to reflect any new settings. Search will continue to work but new content won't be indexed until all existing content has been re-indexed.'

**Comment by jenlampton** [comment]
On the search settings form:
![screen shot 2015-11-05 at 3 08 11 pm](https://cloud.githubusercontent.com/assets/397895/10984559/0f14ccc6-83cf-11e5-9665-5840b07b046a.png)

On the confirm form:
![screen shot 2015-11-05 at 3 08 46 pm](https://cloud.githubusercontent.com/assets/397895/10984576/296cd168-83cf-11e5-80f6-29c13b1b7051.png)

**Comment by docwilmot** [comment]
Not liking "Invalidate", its a bit too techy I feel. Personally I think "Re-index" works fine if its in red and there's a confirm form that threatens hell and damnation if you click it.

**Comment by jenlampton** [comment]
Re-index makes people think that after they click the button their site will be re-indexed. But that's actually the opposite of what happens. After the button is clicked the amount of their site that's indexed drops to 0.  This is a huge, huge usability fail.

What actually happens when that button is clicked is that the site is scheduled for reindexing (the reindexing will happen at some arbitrary time in the future - sometimes that's a very long time!)

The button should convey what will happen when you click it - not what will happen at some arbitrary time in the future. 

Do you have a better suggestion than invalidate?

**Comment by serundeputy** [comment]
queue for re-indexing *shrug*

If you wish to re-index immediately visit  [/admin/config/system/cron] (/admin/config/system/cron)  and run cron until full index is complete.

**Comment by docwilmot** [comment]
Ok, I thought the concern here was a performance issue - I thought clicking that button would slow your site or something, not that the button text was inaccurate. If we're going for accuracy, then I prefer delete over invalidate, or perhaps "Clear Site Index". 

I can't think of a one word which would say both that "you are going to do a bad thing and delete your index" and simultaneously say "it will be rebuilt later". Even "queue for re-indexing" suggests nothing will happen now, which as @jenlampton points out is inaccurate. I think a simple "Delete" or "Clear" with some description text right on the same page which explains that it will be re-indexed on next cron.

While we're at it, why does Backdrop delete the index immediately, but only re-index later? Why not change so on the next cron run the site index is dropped *then* and re-indexed immediately? Then any wording would be OK then.

**Comment by quicksketch** [comment]
> I think a simple "Delete" or "Clear" with some description text right on the same page which explains that it will be re-indexed on next cron.

I think that @jenlampton's use of the word "Invalidate" (though techy) is definitely the most accurate. You're not actually deleting anything, search continues to work. You're just saying that the search index should be replaced with fresher items as soon as enough cron runs happen.

> While we're at it, why does Backdrop delete the index immediately, but only re-index later? Why not change so on the next cron run the site index is dropped then and re-indexed immediately? Then any wording would be OK then.

Totally agree that this would be a much better solution. Though dropping the search index entirely isn't quite as good as invalidating it all and then immediately rebuilding (via a batch operation with progress bar). Then search continues to work the entire time and if any errors occur, the existing index continues to be in place.

**Comment by klonos** [comment]
Can we implement some sort of "temporary" index while the new indexing happens? I mean, we keep using the old search index while we build a new (in the background, on cron runs) and once the new is 100% (or so), then we replace the old with the new and only after that has happened we delete the old index. Is that what you were talking about? ...cause that's what came to mind when I read @docwilmot's comment.

**Comment by docwilmot** [comment]
OK then "Invalidate" it is. Separate issue to change how this works?

**Comment by klonos** [comment]
> ...Separate issue to change how this works?

#1337 :wink:

**Comment by sutibun** [comment]
I'm actually a perfect example. I never used the search module in Drupal until encountering it first with Backdrop (since it was on by default) When I first explored this admin page, *I was tempted* to click "Re-index"... especially since it was at the top and looked friendly.

Anyway, "Invalidate" sounds way too harsh. I like the sound of "Clear Site Index"

I don't really have a good alternative but how about 

- Using a synonym for "invalidate"? At the top of my head can only think of "void", "nullify", or maybe even "repair"? OR
- Hiding the "Reindex site" button under a collapsed section called "Advanced". I really don't know since I never used search in Drupal/Bd but is "reindexing site" something that will be used often by most users such that it needs to be visible all the way on the top of the page?

**Comment by jenlampton** [comment]
> I like the sound of "Clear Site Index"

The problem is that your site index is not cleared when you push that button. What happens is the "Percent indexed" number that you see on the page drops to zero, but nothing else changes!  Your site index is not dropped (it's still there, search still works). But now that you have clicked this button, all of the content in that index will slowly be replaced, one item at a time.

**Comment by docwilmot** [comment]
Invalidate is sufficiently descriptive, accurate and scary. Let's keep that.

**Comment by jenlampton** [comment]
Okay, i rebased my PR and while I was in there I moved the "scary" settings from the top of the search page to the bottom.
![screen shot 2016-01-01 at 5 59 47 pm](https://cloud.githubusercontent.com/assets/397895/12072898/b14a6578-b0b1-11e5-90ee-59ce3cac280d.png)

**Comment by quicksketch** [comment]
Looks great! Merged into 1.x for 1.3.0. We don't want to change strings unnecessarily in bugfix releases (breaking translations) so we'll leave this to be in 1.3.0 only.

**Comment by alanmels** [comment]
I like the neatness of the first screenshot that @jenlampton proposed on https://github.com/backdrop/backdrop-issues/issues/1321#issuecomment-154225281. Placing the same warning message:

> The search index is not cleared immediately when invalidated, instead it will be systematically updated during cron runs. Searching will continue to work but new content won't be indexed until all existing content has been re-indexed.

twice on the search settings form and then again on the confirm page is little too excessive. I understand the initial concern some people could press the button without realizing the consequences, however they are in fact not so scary, because (1) no data is lost anyway and then (2) those who decided to press the button should really see the warning on the confirm page. Was this concern really justified in order to place the same text at two places?
